worker/firewaller/firewaller.go

```diff
@@ -5,6 +5,7 @@ package firewaller
 
 import (
 	"io"
```

Comment:
I got to here, but I want to give this part more focused attention, I will try to look at this a bit later. (dinner time)